### PR TITLE
[DeleteFilesV1] Migration of DeleteFilesV1 to node10

### PR DESCRIPTION
**Pull request by EzzhevNikita**
**Task name**: DeleteFilesV1

**Description**: Updated dependencies; Updated execution handler; Replaced deprecated methods MochaDone => Mocha.Done; Removed typings; Fixed typo path => curPath in removeFolder method in tests; Removed legacy tests;

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) Y

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

**[Node 6 backup branch](https://github.com/microsoft/azure-pipelines-tasks/tree/node6hotfixes/DeleteFilesV1)**

### CODE REVIEW

File: Tasks/DeleteFilesV1/package.json [hunk 1]
```diff
@@ -17,6 +17,11 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "azure-pipelines-task-lib": "^2.9.3"
+    "@types/node": "^10.17.0",
+    "@types/mocha": "^5.2.7",
+    "azure-pipelines-task-lib": "3.0.6-preview.0"
```
Comment (egor-bryzgalov) [inline Tasks/DeleteFilesV1/package.json:22]:
```suggestion
    "azure-pipelines-task-lib": "^3.0.6-preview.0"
```

Comment (EzzhevNikita) [reply Tasks/DeleteFilesV1/package.json]: Thanks, fixed

**Review by egor-bryzgalov — APPROVED** [review]
LGTM. Thanks!

**Review by anatolybolshakov — APPROVED** [review]

File: Tasks/DeleteFilesV1/Tests/L0.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ const testRoot = path.join(__dirname, 'test_structure');
 const removeFolder = function(curPath) {
     if (fs.existsSync(curPath)) {
         fs.readdirSync(curPath).forEach((file, index) => {
-        const newPath = path.join(path, file);
+        const newPath = path.join(curPath, file);
```
Comment (anatolybolshakov) [inline Tasks/DeleteFilesV1/Tests/L0.ts:11]: Is this intentional change? Is it some fix for test here?

**Review by sdobrodeev — APPROVED** [review]

File: Tasks/DeleteFilesV1/Tests/L0.ts (hunk 2 above)
Comment (EzzhevNikita) [reply Tasks/DeleteFilesV1/Tests/L0.ts:11]: Yes, it seems to be a typo that did not cause build errors before or other errors during test execution (it strange to my mind). But after bumping of the dependencies versions it started throwing an error during build. And in fact _'path'_ is an imported module and path.join(_path_, file) looks like a meaningless construction and should produce error.

**Comment by alex-peck** [comment]
Can you also delete the entries in the Tests-Legacy folder related to DeleteFiles?

**Comment by EzzhevNikita** [comment]
> Can you also delete the entries in the Tests-Legacy folder related to DeleteFiles?

Legacy tests removed, thank you!

**Review by alex-peck — APPROVED** [review]

**Pull request merged**